src/Google.Cloud.Functions.Framework/IHttpRequestReader.cs

```diff
@@ -0,0 +1,19 @@
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using System;
+
+namespace Google.Cloud.Functions.Framework
+{
+    /// <summary>
+    /// Responsible for reading a strongly typed request object from an
+    /// HTTP request.
+    /// </summary>
+    public interface IHttpRequestReader<TReq>
```

Comment:
It's an interesting thought, CancellationTokens are used to abort an operation if it's long running and the HTTP request dies? I'd say I'm also ambivalent. 

The cases I can imagine where it might be useful would be borderline applications of formatters i.e. augmenting the request with data fetched from a Cloud Storage bucket (or some similarly slow and expensive network operation you'd want to be able to cancel). In the typical case request decoding / response encoding should be so fast I can't imagine too much value.

WDYT?

Reply:
I agree that usually it'll be fast, in terms of processing. If a large result is generated, then it would be good to abort that early, but I'd expect that if the request is aborted, that output stream will become invalid anyway.

Let's go without for now... introducing it later would be a breaking change, but we can always introduce other interfaces to implement, and work things out that way.

Reply:
There's no need to explicitly pass the cancellation token. HttpRequest and HttpResponse both have access to the HttpContext, which in turn has the `CancellationToken HttpContext.RequestAborted` property.

The alternative I would consider for the reader and the writer is to actually pass the HttpContext directly. That makes it clearer that the whole thing is available for using in serialization/deserialization. There are other elements of HttpContext that may be useful for deserialization, like the Items collection.

Reply:
Ah, hooray. Good to know we don't need to pass anything else.

But I prefer keeping the request/response directly - it emphasizes what the code is expected to work with. (It would be relatively unusual for a request reader to need to reach into the context, for example.) So long as they *can* do it, I think that's fine.